### PR TITLE
mt7621: add support for PanguBox PGB-M1 reference board

### PR DESCRIPTION
**Pull request by KyrinWrt**
PanguBox PGB-M1 reference board

Specifications:

CPU:    MediaTek MT7621AT@880MHz
DRAM:   512Mb DDR3
Flash:  32Mb SPI-NOR
WAN:    1x
LAN:    4x
WLAN:   MT7905DAN(MAC) + MT7975DN(RF)
USB:    1x 3.0
Button: 1x reset
LEDs:   a lot
4G module:Sierra Wireless AirPrime EM7355
Slot：1x sim card slot

Base configuration:
    1.Run the following command to open the OpenWrt Graphical configuration interface.
        make menuconfig
    2.Selecting a Hardware Platform
        Location:
            ->Target System(MediaTek Ralink MIPS)
            ->Subtarget(MT7621 based boards)
            ->Target Profile(PanguBox PGB-M1)
    3.Selecting LuCI package
        (1)LuCI interface with Uhttpd as Webserver
        Location:
            ->LuCI
                -> 1.Collections
                <*> luci
           Location:
        (2)Select the language for the LuCI Web display
            ->LuCI
                ->2.Modules
                ->Translations
                <*>English(en)
                <*>Chinese Simplified(zh_Hans)
                ......

Build the project：

    Run the following command in the project root directory.
    make V=99

Obtaining firmware:
    The firmware is stored in the following directory.
    ${HOME}/openwrt/bin/targets/ramips/mt7621

Installation:

    1. Upgrade the firmware using tftp(PGB-M1 is provided by default)
    Note:
        The development board must be in the same network segment as the PC.
    (1)Use a network cable to connect the LAN interface of the development
        board to the PC network port.
    (2)Set the IP address of the development board to a static IP address.
        For example: 192.168.1.1
    (3)Set the IP address of the TFTP server IP address.
        For example: 192.168.1.100
    (4)Reboot the board.
    (5)Press key 2 on your keyboard to start burning the firmware.
    (6)Input device IP (192.168.1.1) ==:<device ip>
    (7)Input server IP (192.168.1.100) ==:<server ip>
    (8)Input Linux Kernel filename () ==:<kernel image>
    Example:
        ......
        [After the development board is powered on, please press no. 2 to select the burning mode of the firmware.]
        ......
        2: Load Firmware then write to Flash via TFTP
         Warning!! Erase Linux in Flash then burn new one. Are you sure? (Y/N)
         Please Input new ones /or Ctrl-C to discard
         Input device IP (192.168.1.1) ==:192.168.1.1
         Input server IP (192.168.1.100) ==:192.168.1.100
         Input Linux Kernel filename () ==:openwrt-ramips-mt7621-pangubox_pgb-m1-squashfs-sysupgrade.bin

         NetLoop,call eth_halt !
         NetLoop,call eth_init !
         Ralink GMAC initializing
         TFTP from server 192.168.1.100; our IP address is 192.168.1.1
         Filename 'openwrt-ramips-mt7621-pangubox_pgb-m1-squashfs-sysupgrade.bin'.

         TIMEOUT_COUNT=10,Load address: 0x80100000
         Loading: Got ARP REQUEST, return our IP
         Got ARP REQUEST, return our IP
         ......

    2. Use Web Browser Upgrade the Firmware

    (1)Use a network cable to connect the LAN interface of the development
        board to the PC network port.
    (2)Set the IP address of the computer to a static IP address.
        For example, 192.168.1.100.
    (3)Restart the development board and press the reset button for about 5 seconds.
    (4)In this case, the following interface is displayed in logs
       printed by the serial port.
    Example:
        ______________________________________________________
        |                                                    |
        |            HTTPD Recovery Module v3.0              |
        |                                                    |
        | Note:Please Use Web Browser Upgrade the Firmware ! |
        |                                                    |
        |          Copyright 2020 PandoraBox Team            |
        |____________________________________________________|

        [kernel]uOS Version: v2.0
        [kernel]PWM-LED Thread started!!
        [kernel]Task Thread started!!
        [TCP/IP]ETH0 MAC:00:20:76:00:00:1E

        [TCP/IP]Max ReceiveBuffer:1492!
        [TCP/IP]IP:192.168.1.1
        [TCP/IP]Netmask:255.255.255.0
        [TCP/IP]Default Router:0.0.0.0
        [DHCPD]Server IP:192.168.1.1
        [DHCPD]IP Pool:192.168.1.100-192.168.1.150

        Ralink GMAC initializing
        [kernel]Enter uOS core task!
        waiting ......
    (5)In the browser, access the following IP address
        192.168.1.1
        Then, follow the instructions on the upgrade page.

### CODE REVIEW

**Comment by adschm** [comment]
https://openwrt.org/docs/guide-developer/frequent-pr-mistakes#do_not_multi-post_or_re-post_prs

File: target/linux/ramips/dts/mt7621_pangubox_pgb-m1.dts [hunk 1]
```diff
@@ -0,0 +1,158 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "mt7621.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible =  "PanguBox,PGB-M1","mediatek,mt7621-soc";
```
Comment (adschm) [inline target/linux/ramips/dts/mt7621_pangubox_pgb-m1.dts:9]: lower case

Comment (adschm) [reply target/linux/ramips/dts/mt7621_pangubox_pgb-m1.dts:9]: And please tidy up spaces

File: target/linux/ramips/dts/mt7621_pangubox_pgb-m1.dts [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+		label-mac-device = &gmac0;
+	};
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+		bootargs = "console=ttyS0,115200n8";
+	};
+
+	leds {
+		compatible = "gpio-leds";
```
Comment (adschm) [inline target/linux/ramips/dts/mt7621_pangubox_pgb-m1.dts:26]: Empty line after gpio-leds

File: target/linux/ramips/dts/mt7621_pangubox_pgb-m1.dts [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+		led_power_green: sys {
+			label = "blue:sys";
```
Comment (adschm) [inline target/linux/ramips/dts/mt7621_pangubox_pgb-m1.dts:28]: Honestly? led_power_green for blue:sys?

I'll wait until this whole thing is ready for presentation.

**Review by adschm — CHANGES_REQUESTED** [review]
This still needs a lot of cleanup before it makes sense to review it.

Also have a look at the following ressources (again) for fixing formal issues:
https://openwrt.org/submitting-patches
https://openwrt.org/docs/guide-developer/device-support-policies
https://openwrt.org/docs/guide-developer/frequent-pr-mistakes

File: target/linux/ramips/dts/mt7621_pangubox_pgb-m1.dts (hunk 3 above)
Comment (TorvaldsRodriguez) [reply target/linux/ramips/dts/mt7621_pangubox_pgb-m1.dts:28]: I'm sorry, I will check again before submitting next time.I am trying to commit pr to OpenWRT and may make some silly mistakes.Thank you for pointing out my problems in your busy schedule.

File: target/linux/ramips/dts/mt7621_pangubox_pgb-m1.dts (hunk 1 above)
Comment (TorvaldsRodriguez) [inline target/linux/ramips/dts/mt7621_pangubox_pgb-m1.dts:9]: compatible = "pangubox,pgb-m1", "mediatek,mt7621-soc";

Comment (TorvaldsRodriguez) [reply target/linux/ramips/dts/mt7621_pangubox_pgb-m1.dts:9]: compatible = "pangubox,pgb-m1", "mediatek,mt7621-soc";

Comment (TorvaldsRodriguez) [reply target/linux/ramips/dts/mt7621_pangubox_pgb-m1.dts:9]: compatible = "pangubox,pgb-m1", "mediatek,mt7621-soc";

**Review by KyrinWrt — COMMENTED** [review]
compatible = "pangubox,pgb-m1", "mediatek,mt7621-soc";

**Pull request closed without merging**